apps/sms/js/activity_handler.js

```diff
@@ -294,70 +293,56 @@ var ActivityHandler = {
       };
 
     if (Compose.isEmpty()) {
-      this.launchComposer(activity);
-    } else {
-      // ask user how should we do
-      ActivityHandler.displayUnsentConfirmation(activity);
+      return this.launchComposer(activity);
     }
+
+    // ask user how should we do
+    ActivityHandler.displayUnsentConfirmation(activity);
```

Comment:
If Jenny and Fang agree to change confirm dialog styling, I'd probably get rid of this method in favor of plain ```Utils.confirm``` and then I can return its promise-based result too :)